core/src/main/java/hudson/cli/CLIAction.java

```diff
@@ -128,7 +127,17 @@ public HttpResponse doWs(StaplerRequest req) {
         }
         if (ALLOW_WEBSOCKET == null) {
             final String actualOrigin = req.getHeader("Origin");
-            final String expectedOrigin = StringUtils.removeEnd(StringUtils.removeEnd(Jenkins.get().getRootUrlFromRequest(), "/"), req.getContextPath());
+
+            String o = Jenkins.get().getRootUrlFromRequest();
+            String removeSuffix1 = "/";
+            if (o.endsWith(removeSuffix1)) {
+                o = o.substring(0, o.length() - removeSuffix1.length());
+            }
+            String removeSuffix2 = req.getContextPath();
+            if (o.endsWith(removeSuffix2)) {
+                o = o.substring(0, o.length() - removeSuffix2.length());
+            }
+            final String expectedOrigin = o;
```

Comment:
The extra `o` variable seems like an unnecessary layer of indirection for the reader. Why not just name it `expectedOrigin` to begin with and remove this line.

Reply:
`expectedOrigin` is a local variable used in lambda, it must be a `final` or `effectively final`.

Of course we can set `o` as another name for readability, but I think it is not very necessary as a temporary variable.